class/authenticator.ts

```diff
@@ -61,11 +85,43 @@ export class Authenticator implements IAuthenticator {
       this.entropy = _entropy;
       this.secret = mnemonic || bytesToMnemonic(buffer);
       this.publicKey = privateKeyToPublicKey(this.privateKey);
+      this.keyPair = ECPair.fromPrivateKey(this.privateKey, {
+        network: config.network,
+      });
     } catch (_) {
       throw new Error(i18n.wallets.errors.invalidPrivateKey);
     }
   }
 
+  async signAndFinalizePSBT(encodedPSBT: string): Promise<FinalizedPSBT> {
+    let tx, fee;
+    let vaultTxType = VaultTxType.Recovery;
+    try {
+      ({ tx, fee } = signer.signAndFinalizePSBT(encodedPSBT, [this.keyPair], vaultTxType));
+    } catch (_) {
+      try {
+        vaultTxType = VaultTxType.Instant;
+        ({ tx, fee } = signer.signAndFinalizePSBT(encodedPSBT, [this.keyPair], vaultTxType));
+      } catch (e) {
+        throw new Error('Unable to sign tx with authenticator: ' + e);
```

Comment:
e.message probably
tip: in js there is a better way to merge strings https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals